### PR TITLE
Docs: adding relative links

### PR DESCRIPTION
**Pull request by s-makin**
## Proposed Commit Message
```
docs: changed all internal links to be relative

In preparation for migrating the docs to RTD.com, identified all 
links that point to the RTD.org links and changed them to be 
relative so that it doesn't matter where they are rendered. 

Also fixed the sphinx-design extension - although it's not used
yet, I will be using it to create the Diataxis home page sections.
```

### CODE REVIEW

File: doc/rtd/topics/integration_tests.rst [hunk 1]
```diff
@@ -59,7 +59,7 @@ Cloud Interaction
 Cloud interaction happens via the
 `pycloudlib <https://pycloudlib.readthedocs.io/en/latest/index.html>`_ library.
 In order to run integration tests, pycloudlib must first be
-`configured <https://pycloudlib.readthedocs.io/en/latest/configuration.html#configuration>`_.
+`configured <topics/configuration.rst>`_.
```
Comment (s-makin) [inline doc/rtd/topics/integration_tests.rst:62]: I just noticed that this wasn't pointing to the current hosted docs - do I need to revert this change?

**Review by blackboxsw — CHANGES_REQUESTED** [review]
Thanks @s-makin for this PR!  Changeset looks good. A couple changes
 - revert the pycloudlib related link change to point externally (not relative link).
 - define a new anchor .. _contributing on the contributing.rst page and use :ref: on index.rst
 - minor fixup of the git branch to ensure we can merge without merge conflicts

The git commit history on this branch looks like a couple of git merge main process things got in the way a bit (preventing this from being merged back into main without conflicts). I can see a list of commits listed in github UI on the "Commits" tab that shows "5" commits, but it looks like your changeset should only be 2 commits so this is that indication that something is amiss with the PR.

I think we can fix this in the following way in your local repo:
```
# 1. Fixup per review comments above
git commit -am 'address review comments' 

# 2. sync your local main branch to upstream/main (generally we want to do this before embarking on a new PR to make sure you are basing your new PR off of what was most recently landed in upstream/main anyway)

git checkout main;  git pull  # ensure you see "Already up to date" telling your your local 'main' equals upstream/main

# 3. Create a 2nd branch to track clean upstream main into which we will grab full changeset to docs
git checkout upstream/main -B  docs-relative-links-2

# 4. check diff  in the 'doc' directory between your `docs-relative-link` doc and your refreshed `main`
git diff upstream/main docs-relative-links doc > my.patch

# 5. apply this patch to your new local docs-related-links-2 branch
patch -p1 < my.patch

# check the local diff before commiting to docs-relative-links-2 branch
git diff

# commit with substantive change commit message for changeset
git commit -a

# double check changes against your previous branch
git diff docs-relative-links docs-relative-links-2 doc

# If that diff looks good you can push it directly over your existing PR branch by renaming docs-relative-links-2 to docs-relative-links during push
git push origin docs-relative-links-2:docs-relative-links 
```

File: doc/rtd/topics/datasources/ovf.rst [hunk 2]
```diff
@@ -11,4 +11,36 @@ transport.
 For further information see a full working example in cloud-init's
 source code tree in doc/sources/ovf
 
+Configuration
```
Comment (blackboxsw) [inline doc/rtd/topics/datasources/ovf.rst:14]: @s-makin, was this documentation add intended in this branch?

**Review by blackboxsw — APPROVED** [review]
LGTM! diff is now good.

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -182,14 +181,13 @@ Do these things for each feature or bug
   - Click 'Create Pull Request`
 
 Then, a cloud-init committer will review your changes and
-follow up in the pull request.  Look at the `Code Review Process`_ doc
+follow up in the pull request.  Look at the :ref:`Code Review Process<code-review-process>` doc
```
Comment (blackboxsw) [inline CONTRIBUTING.rst:184]: `tox -re doc` fail in CI and locally. I think because your anchor name is hyphenated instead of underscore delmited.

File: doc/rtd/topics/code_review.rst [hunk 4]
```diff
@@ -1,3 +1,5 @@
+.. _code-review-process:
```
Comment (blackboxsw) [inline doc/rtd/topics/code_review.rst:1]: I think this needs underscores not hyphens

File: CONTRIBUTING.rst [hunk 5]
```diff
@@ -17,15 +19,12 @@ Before any pull request can be accepted, you must do the following:
 * Add your Github username (alphabetically) to the in-repository list that we use
   to track CLA signatures:
   `tools/.github-cla-signers`_
-* Add or update any `unit tests`_ accordingly
-* Add or update any `integration tests`_ (if applicable)
+* Add or update any :ref:`unit tests<testing>` accordingly
+* Add or update any :ref:`integration tests` (if applicable)
```
Comment (blackboxsw) [inline CONTRIBUTING.rst:23]: integration tests needs and underscore instead of to find the right anchor.
```suggestion
* Add or update any :ref:`integration_tests` (if applicable)
```

**Review by blackboxsw — APPROVED** [review]
Thanks for fixing this up @s-makin.  A couple of minor nits I've commented on that broken tox -re docs.
Here's the consolidated diff of my suggestions and we are good to go here. I'll merge them as they shouldn't be a contentious set of changes.
```diff
diff --git a/CONTRIBUTING.rst b/CONTRIBUTING.rst
index 7f1c87da8..e9e254998 100644
--- a/CONTRIBUTING.rst
+++ b/CONTRIBUTING.rst
@@ -20,7 +20,7 @@ Before any pull request can be accepted, you must do the following:
   to track CLA signatures:
   `tools/.github-cla-signers`_
 * Add or update any :ref:`unit tests<testing>` accordingly
-* Add or update any :ref:`integration tests` (if applicable)
+* Add or update any :ref:`integration_tests` (if applicable)
 * Format code (using black and isort) with `tox -e do_format`
 * Ensure unit tests and linting pass using `tox`_
 * Submit a PR against the `main` branch of the `cloud-init` repository
@@ -76,7 +76,7 @@ Follow these steps to submit your first pull request to cloud-init:
     git remote add upstream git@github.com:canonical/cloud-init.git
     git push origin main
 
-* Read through the cloud-init `Code Review Process`_, so you understand
+* Read through the cloud-init :ref:`Code Review Process<code_review_process>`, so you understand
   how your changes will end up in cloud-init's codebase.
 
 * Submit your first cloud-init pull request, adding your Github username to the
@@ -181,7 +181,7 @@ Do these things for each feature or bug
   - Click 'Create Pull Request`
 
 Then, a cloud-init committer will review your changes and
-follow up in the pull request.  Look at the :ref:`Code Review Process<code-review-process>` doc
+follow up in the pull request.  Look at the :ref:`Code Review Process<code_review_process>` doc
 to understand the following steps.
 
 Feel free to ping and/or join ``#cloud-init`` on Libera irc if you
diff --git a/doc/rtd/topics/code_review.rst b/doc/rtd/topics/code_review.rst
index 5b2dd4f98..2be78ed58 100644
--- a/doc/rtd/topics/code_review.rst
+++ b/doc/rtd/topics/code_review.rst
@@ -1,6 +1,5 @@
-.. _code-review-process:
+.. _code_review_process:
 
-*******************
 Code Review Process
 *******************
 
```
